presentation.js

```diff
@@ -198,15 +199,19 @@ function toggleChords(touch=false) {
 }
 
 function searchDatabase(){
-  var text = $('#searchbox input').val().replace(/[^a-zA-Z0-9\s:-]/g, '');
+  var text = $('#searchbox input').val();
   if(jQuery.trim(text).length < 2){
     $('#searchresults').empty();
     return;
   }
 
-  var re = new RegExp(text, 'i');
+  var re = new RegExp('(' + text + ')([^\0\n]*)', 'i');
```

Comment:
Do we need to check for both '\n' and '\0'?  We don't need to make it work with prior databases at this point.

Reply:
My recollection is that I had to check for both because we're now using `\0` to separate between song parts but that a `\n` was still used between lines in the song chunk.  But as I write that I'm realizing that the only reason we'd have `\n` in the song chunks is because of my other regex change to only convert two or more consecutive whitespace to a space.  

... and now I'm thinking that that change is wrong since we want to convert ALL white space to the normal space character so that when we search we don't care what kind of white space separates words.  OTOH, do we want searches to line wrap?  There's times that could be handy, but I'm not fully certain ...  it would also bring up new difficulties (currently we show from start of line to end of line when displaying search results).  I need to think about this more ... what are your thoughts?

This will need change whichever way we go, I think how it is currently we'll strip new lines if the end of the previous line has a space character.

Reply:
Search line wrapping already currently works in the app.

In presentation, I think it may be more bother than it's worth to enable line wrapped search display.  I think I've had times where I've searched a phrase, it didn't work, so I tried something else, or took a shorter snippet and that did work - somewhat annoying but not bad.

-- Randomish processing --
Another possibility I've been playing around with in my head regarding search for the app is to enable more of a fuzzy search type feature.  Right now it's not easy to search for a song with two categories, or an author and a category because the search matches exactly to the generated search string.  Would be nice to be able to do a query like: `a:"Bruce Hempel" c:Praise "Great is the Lord"` currently that is pretty much impossible.  

I was thinking this would reduce the need for an answer here on line wrapping, but now realize it doesn't really change it.
-- End randomish processing --

A wrinkle in considering all line wrapping: Currently the search string is a list of all unique lines in a song, if the song has repeated lines they're not in the search string, so a line wrap would not always match 100% of the time when it should.  For example if a repeated line has a different following line.

I think I fall on the let's not worry about line wrapping side of things.  We can change it later if we want to.

Reply:
I think if we decide to keep the search line wrapping as is (e.g. disabled in presentation mode), then the code is ok as is - we need both the breaking on `\0` (for between title, categories, etc) and `\n` (for between lines in the song).